PSModuleDevelopment/functions/utility/Show-PSMDSyntax.ps1

```diff
@@ -0,0 +1,206 @@
+﻿function Show-PSMDSyntax {
+    <#
+    .SYNOPSIS
+        Validate or show parameter set details with colored output
+
+    .DESCRIPTION
+        Analyze a function and it's parameters
+
+        The cmdlet / function is capable of validating a string input with function name and parameters
+
+    .PARAMETER CommandText
+        The string that you want to analyze
+
+        If there is parameter value present, you have to use the opposite quote strategy to encapsulate the string correctly
+
+        E.g. for double quotes
+        -CommandText 'Import-D365Bacpac -ImportModeTier2 -SqlUser "sqladmin" -SqlPwd "XyzXyz" -BacpacFile2 "C:\temp\uat.bacpac"'
+        
+        E.g. for single quotes
+        -CommandText "Import-D365Bacpac -ExportModeTier2 -SqlUser 'sqladmin' -SqlPwd 'XyzXyz' -BacpacFile2 'C:\temp\uat.bacpac'"
+
+    .PARAMETER Mode
+        The operation mode of the cmdlet / function
+
+        Valid options are:
+        - Validate
+        - ShowParameters
+
+    .PARAMETER IncludeHelp
+        Switch to instruct the cmdlet / function to output a simple guide with the colors in it
+
+    .EXAMPLE
+        PS C:\> Show-PSMDSyntax -CommandText 'Import-D365Bacpac -ImportModeTier2 -SqlUser "sqladmin" -SqlPwd "XyzXyz" -BacpacFile2 "C:\temp\uat.bacpac"' -Mode "Validate"
+
+        This will validate all the parameters that have been passed to the Import-D365Bacpac cmdlet.
+        All supplied parameters that matches a parameter will be marked with an asterisk.
+        
+    .EXAMPLE
+        PS C:\> Show-PSMDSyntax -CommandText 'Import-D365Bacpac' -Mode "ShowParameters"
+
+        This will display all the parameter sets and their individual parameters.
+
+    .NOTES
+        Author: Mötz Jensen (@Splaxi)
+
+#>
+    [CmdletBinding()]
+    
+    param (
+        [Parameter(Mandatory = $true, Position = 1)]
+        [string] $CommandText,
+
+        [Parameter(Mandatory = $true, Position = 2)]
+        [ValidateSet('Validate', 'ShowParameters')]
+        [string] $Mode,
```

Comment:
Why is this parameter mandatory?
Wouldn't a default value of `'Validate'` be more convenient for the user?
Also: Why include modes at all?
I have yet to see a meaningful difference between the modes.

Reply:
I agree on the default value, that it should be validate.

The main reason for the 2 modes is maybe a bit vague, but the validate shows asterisk all around the output. From a user perspective I thought that it might confuse people the first few times, until the understood the function. Let's visit this when we have nailed the rest.